src/core/metadata/qgslayermetadataformatter.cpp

```diff
@@ -0,0 +1,217 @@
+      }
+      if ( ! oneAddress.administrativeArea.isEmpty() )
+      {
+        myMetadata += oneAddress.administrativeArea % QLatin1String( "<br />" );
+      }
+      if ( ! oneAddress.country.isEmpty() )
+      {
+        myMetadata += oneAddress.country;
+      }
+      notFirstRow = true;
+    }
+    myMetadata += "</td></tr>\n";
+    i++;
+  }
+  myMetadata += QLatin1String( "</table>\n" );
+  return myMetadata;
+}
+
+QString QgsLayerMetadataFormatter::extentSectionHtml( const QgsLayerMetadata &metadata ) const
+{
+  QString myMetadata = QStringLiteral( "<table class=\"list-view\">\n" );
+  myMetadata += QStringLiteral( "<tr><td class=\"highlight\">" ) % QObject::tr( "CRS" ) % QLatin1String( "</td><td>" ) % metadata.crs().authid() % QLatin1String( "</td></tr>\n" );
+
+  const QgsLayerMetadata::Extent extent = metadata.extent();
+  myMetadata += QStringLiteral( "<tr><td class=\"highlight\">" ) % QObject::tr( "Spatial Extent" ) % QLatin1String( "</td><td>" );
+  QList< QgsLayerMetadata::SpatialExtent > spatialExtents = extent.spatialExtents();
+  bool notFirstRow = false;
+  for ( const QgsLayerMetadata::SpatialExtent &spatialExtent : spatialExtents )
+  {
+    if ( notFirstRow )
+    {
+      myMetadata += QLatin1String( "<br />\n" );
+    }
+    myMetadata += QStringLiteral( "<strong>" ) % QObject::tr( "CRS" ) % QStringLiteral( ": </strong>" ) % spatialExtent.extentCrs.authid() % QStringLiteral( "<br />" );
+    myMetadata += QStringLiteral( "<strong>" ) % QObject::tr( "X Minimum" ) % QStringLiteral( ": </strong>" ) %  qgsDoubleToString( spatialExtent.bounds.xMinimum() ) % QStringLiteral( "<br />" );
+    myMetadata += QStringLiteral( "<strong>" ) % QObject::tr( "Y Minimum" ) % QStringLiteral( ": </strong>" ) %  qgsDoubleToString( spatialExtent.bounds.yMinimum() ) % QStringLiteral( "<br />" );
+    myMetadata += QStringLiteral( "<strong>" ) % QObject::tr( "X Maximum" ) % QStringLiteral( ": </strong>" ) %  qgsDoubleToString( spatialExtent.bounds.xMaximum() ) % QStringLiteral( "<br />" );
+    myMetadata += QStringLiteral( "<strong>" ) % QObject::tr( "Y Maximum" ) % QStringLiteral( ": </strong>" ) %  qgsDoubleToString( spatialExtent.bounds.yMaximum() ) % QStringLiteral( "<br />" );
+    if ( spatialExtent.bounds.zMinimum() || spatialExtent.bounds.zMinimum() )
+    {
+      myMetadata += QStringLiteral( "<strong>" ) % QObject::tr( "Z Minimum" ) % QStringLiteral( ": </strong>" ) %  qgsDoubleToString( spatialExtent.bounds.zMinimum() ) % QStringLiteral( "<br />" );
+      myMetadata += QStringLiteral( "<strong>" ) % QObject::tr( "Z Maximum" ) % QStringLiteral( ": </strong>" ) %  qgsDoubleToString( spatialExtent.bounds.zMaximum() );
+    }
+    notFirstRow = true;
+  }
+  myMetadata += QLatin1String( "</td></tr>\n" );
+  myMetadata += QStringLiteral( "<tr><td class=\"highlight\">" ) % QObject::tr( "Temporal Extent" ) % QLatin1String( "</td><td>" );
+  QList< QgsDateTimeRange > temporalExtents = extent.temporalExtents();
```

Comment:
Same here - store the result here in a const variable. There's more details about this at https://github.com/qgis/QGIS-Enhancement-Proposals/issues/103